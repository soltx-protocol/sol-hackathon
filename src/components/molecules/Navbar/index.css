.navbar {
	width: 240px;
	height: 100vh;
	padding: 18px;
	background-color: var(--neutral-white);
	border-right: 1px solid var(--divider-20);

	& > .icon {
		padding: 6px 10px;
		margin-bottom: 18px;
	}

	& > button {
		display: flex;
		height: 42px;
		width: 100%;
		border-radius: 8px;
		margin-bottom: 4px;
		background-color: var(--neutral-white);
		transition: background-color 0.3s ease-out;

		& > div {
			display: flex;
			justify-content: flex-start;
			align-items: center;
			padding: 0 13px;
			width: 100%;
			height: 100%;
			font-style: normal;
			font-weight: 600;
			font-size: 16px;
			line-height: 24px;
			color: var(--neutral-control-layer);
			transition: color 0.3s ease-out;

			& > svg {
				margin-right: 18px;
			}
		}

		&:hover {
			background-color: #ededfc;
		}

		&.active {
			background-color: #ededfc;

			& > div {
				color: #291ccb;
			}
		}
	}
}
