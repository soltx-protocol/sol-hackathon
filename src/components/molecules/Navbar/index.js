import React from 'react';
import classnames from 'classnames';

import OrcaLogo from 'images/orca.png';
import SoltxLogo from 'images/icon/soltx.inline.svg';
import TradeIcon from 'images/icon/trade.inline.svg';
import TradeActiveIcon from 'images/icon/trade-active.inline.svg';
import WalletIcon from 'images/icon/wallet.inline.svg';
import WalletActiveIcon from 'images/icon/wallet-active.inline.svg';

import styles from './index.css';

export const NAV_SELECT = {
	OVERALL: 'OVERALL',
	WALLET: 'WALLET',
	NONE: 'NONE',
};

const Navbar = ({ className, select = NAV_SELECT.NONE }) => (
	<div className={classnames(styles.navbar, className)}>
		<div className={styles.icon}>
			<img src={OrcaLogo} alt="orca" /> Hackboard
		</div>
		<button
			className={classnames({ [styles.active]: select === NAV_SELECT.OVERALL })}
			type="button"
		>
			<div>{select === NAV_SELECT.OVERALL ? <TradeActiveIcon /> : <TradeIcon />} Overall</div>
		</button>
		<button className={classnames({ [styles.active]: select === NAV_SELECT.WALLET })} type="button">
			<div>
				{select === NAV_SELECT.WALLET ? <WalletActiveIcon /> : <WalletIcon />} Individual Wallet
			</div>
		</button>
		<div className={styles.bottom}>
			<SoltxLogo />
			Created by Soltx
		</div>
	</div>
);

export default Navbar;
