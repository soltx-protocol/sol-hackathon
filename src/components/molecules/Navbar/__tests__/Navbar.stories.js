import React from 'react';
import { storiesOf } from '@storybook/react';
import { select } from '@storybook/addon-knobs/react';

import Navbar, { NAV_SELECT } from 'components/molecules/Navbar';

const stories = storiesOf('molecules/Navbar', module);

stories.add('__interactive', () => (
	<Navbar select={select('select', Object.keys(NAV_SELECT), NAV_SELECT.NONE)} />
));
