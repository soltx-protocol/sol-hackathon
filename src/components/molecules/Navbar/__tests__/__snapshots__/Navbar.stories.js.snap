// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="icon"
    >
      <img
        alt="orca"
        src="test-file-stub"
      />
       Hackboard
    </div>
    <button
      class=""
      type="button"
    >
      <div>
        <test-file-stub />
         Overall
      </div>
    </button>
    <button
      class=""
      type="button"
    >
      <div>
        <test-file-stub />
         Individual Wallet
      </div>
    </button>
    <div
      class="bottom"
    >
      <test-file-stub />
      Created by Soltx
    </div>
  </div>
</DocumentFragment>
`;
